chromium_src/components/search_engines/template_url_prepopulate_data.cc

```diff
@@ -370,9 +482,15 @@ std::vector<std::unique_ptr<TemplateURLData>> GetPrepopulatedEngines(
   if (!t_urls.empty())
     return t_urls;
 
+  int version = kBraveFirstTrackedDataVersion;
```

Comment:
This is why the unit tests are failing. It seems that `GetPrepopulatedEngines` is called before profile is finished being loaded (this used to work before C87 upgrade!). The 2nd run fixes the problem, because AFTER profile is loaded, it sets `kBraveDefaultSearchVersion` as expected.

I can't call the chrome/browser/first_run because that would be a layering violation. Will need to think of a more clever way to check this.

Reply:
@iefremov helped me find a solution; initializing these values in `BraveProfileManager::InitProfileUserPrefs` 😄 

Works great! 👍